opal/mca/common/ucx/common_ucx_wpool.c

```diff
@@ -1186,37 +858,37 @@ opal_common_ucx_winfo_flush(opal_common_ucx_winfo_t *winfo, int target,
     return rc;
 }
 
-
 OPAL_DECLSPEC int
 opal_common_ucx_wpmem_flush(opal_common_ucx_wpmem_t *mem,
                           opal_common_ucx_flush_scope_t scope,
                           int target)
 {
-    _ctx_record_list_item_t *item;
+    _ctx_record_t *ctx_rec;
     opal_common_ucx_ctx_t *ctx = mem->ctx;
     int rc = OPAL_SUCCESS;
 
     opal_mutex_lock(&ctx->mutex);
 
-    OPAL_LIST_FOREACH(item, &ctx->tls_workers, _ctx_record_list_item_t) {
+    OPAL_LIST_FOREACH(ctx_rec, &ctx->ctx_records, _ctx_record_t) {
+        opal_common_ucx_winfo_t *winfo = ctx_rec->winfo;
         if ((scope == OPAL_COMMON_UCX_SCOPE_EP) &&
-                (NULL == item->ptr->endpoints[target])) {
+                (NULL == winfo->endpoints[target])) {
             continue;
         }
-        opal_mutex_lock(&item->ptr->mutex);
-        rc = opal_common_ucx_winfo_flush(item->ptr, target, OPAL_COMMON_UCX_FLUSH_B,
+        rc = opal_common_ucx_winfo_flush(winfo, target, OPAL_COMMON_UCX_FLUSH_B,
```

Comment:
does this function protect the worker with `winfo->mutex`?

Reply:
It does not - I moved the winfo->mutex to include the winfo_flush